content/concepts/did-ddo.md

```diff
@@ -196,21 +196,99 @@ Example:
 }
 ```
 
-During the publish process, file URLs must be encrypted with a respective _Provider_ API call before storing the DDO on-chain. For this an array of strings with one or multiple URLs is what gets encrypted and send to _Provider_:
+During the publish process, file URLs must be encrypted with a respective _Provider_ API call before storing the DDO on-chain. For this an array of objects defining the storage access details are sent. 
+
+
+Type of objects supported :
+
+<table>
+<tr>
+<th>Key</th>
+<th>Description</th>
+<th>Example</th>
+</tr>
+<td>'url'</td>
+<td>Static URL.  Contains url and HTTP method</td>
+<td>
+
+```json
+[
+   {
+      "url": {
+         "url": "https://url.com/file1.csv",
+         "method": "GET"
+      }
+   },
+   {
+      "url": {
+         "url": "https://url.com/file2.csv",
+         "method": "POST"
+      }
+   }
+]
+```
+
+</td>
+</tr></table>
+
+First class integrations supported in the future :
+
+<table>
+<tr>
+<th>Key</th>
+<th>Description</th>
+<th>Example</th>
+</tr>
+<tr>
+<td>"ipfs"</td><td>IPFS files</td>
+<td>
+
+ ```json
+[ 
+  { "ipfs": { "hash": "XXX"}}
+]
+```
+
+</td>
+
+<tr><td>"filecoin"</td><td>Filecoin storage</td><td>&nbsp;</td></tr>
+<tr><td>"arwave"</td><td>Arwave</td><td>&nbsp;</td></tr>
+<tr><td>"storj"</td><td>Storj</td><td>&nbsp;</td></tr>
+<tr><td>"sql"</td><td>Sql connection, dataset is generated by a query</td><td>&nbsp;</td></tr>
+</table>
+
+
+A service can contain multiple files, using multiple storage types.
+
+Example:
 
 ```json
-["https://url.com/file1.csv", "https://url.com/file2.csv"]
+[
+   {
+      "url":{
+          "url": "https://url.com/file1.csv",
+          "method": "GET"
+      }
+   },
+   {
+      "ipfs": {
+         "hash": "XXXX"
+      }
+   }
```

Comment:
Not sure if can find a more generic way of representing the file info object and have some proper typings for it. As it looks like it would be really though since we are going to support storage types like ipfs or sql

Reply:
or we could have something like:
```json
[
      {       "type": "url",
               "url":  "https://xx",
               "method": "POST"
     }, 
      {       "type": "ipfs",
               "hash":  "XXXX"
     }
]
```

it would make typings more clear

Reply:
agree

Reply:
I'm also opting for the structure proposed above, easier to extend and we can have method/has/..etc optional in the typings

Reply:
changed to match the last proposed structure